browser/ui/webui/brave_webui_source.cc

```diff
@@ -1118,3 +1126,33 @@ content::WebUIDataSource* CreateAndAddWebUIDataSource(
   content::WebUIDataSource::Add(Profile::FromWebUI(web_ui), data_source);
   return data_source;
 }
+
+void AddBackgroundColorToSource(content::WebUIDataSource* source,
+                                Profile* profile) {
+  // Get the color provider for the currect theme supplier +
+  // native theme combination. We only really consider
+  // theme suppliers that will be relevant to the color
+  // we need.
+  CustomThemeSupplier* theme_supplier;
+  ui::NativeTheme* native_theme;
+  if (profile->IsIncognitoProfile() || profile->IsTor() ||
+      profile->IsGuestSession()) {
+    theme_supplier = new BravePrivateWindowThemeSupplier(!profile->IsTor());
```

Comment:
this causes leak

Reply:
Maybe you already tried below. Hmm... it doesn't work for private(tor).
```
void AddBackgroundColorToSource(content::WebUI* web_ui,
                                content::WebUIDataSource* source) {
  auto* web_contents = web_ui->GetWebContents();

  // Get the css value for the NTP background
  SkColor ntp_background_color =
      web_contents->GetColorProvider().GetColor(kColorNewTabPageBackground);
  std::string ntp_background_color_css =
      ui::ConvertSkColorToCSSColor(ntp_background_color);
  // Set to a template replacement string that can be inserted to the
  // html.
  source->AddString("backgroundColor", ntp_background_color_css);
}
```

Reply:
IMO, just putting colors directly to |source| seems fine instead of using `ColorProvider`
because we know all conditions and it's complex to use it here.

Reply:
That's a fair point. Yes I tried a variety of ways and this was the only one that worked, aside from repeating the values. But yes I could just use the c++ constants at least.

Reply:
I wonder if we should fix `web_contents->GetColorProvider()` though, at some point!

Reply:
> I wonder if we should fix `web_contents->GetColorProvider()` though, at some point!

I didn't dig much but we could do it via `WebContents::WebSetColorProviderSource()`
This could be a good f/u.